changelog.d/979.fix.md

```diff
@@ -0,0 +1 @@
+Replaced all usages of `timestamp_nanos` with `timestamp_nanos_opt` to avoid panics when the timestamp is out of range.
```

Comment:
```suggestion
`to_unix_timestamp`, `to_float`, and `uuid_v7` can now return an error if the supplied timestamp is unrepresentable as a nanosecond timestamp. Previously the function calls would panic.
```

The changelogs should have the user in mind as the audience. I also think this is a breaking change, no? We'll want to document how to migrate.

Reply:
Right, marking this PR as a draft for now. I will try to find some propose a migration plan. I need to refresh my memory on the deprecation process as well.

Reply:
We provide linters for such changes. In this way we have eliminated the deprecation of to_timestamp. 
I think there is no problem in this case, because the vector during validation will report that to_unix_timestamp returns an error, and this is not handled

Reply:
Any plans to release the fix?

Reply:
In this case, we would have to document that to_unix_timestamp (and uuid_v7) are now fallible. And then users would have to update the VRL source in their configs when updating their Vector version.